activerecord/lib/active_record/connection_adapters/connection_specification.rb

```diff
@@ -17,6 +17,37 @@ def to_hash
         @config.merge(name: @name)
       end
 
+      class ConnectionConfigurations < SimpleDelegator #:nodoc:
```

Comment:
Isn't inheriting from SimpleDelegator effectively the same thing as inheriting from Hash?

Maybe we should just restrict the allowed interface to `[]`?